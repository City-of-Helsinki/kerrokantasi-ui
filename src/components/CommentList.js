import React from 'react';
import {injectIntl, FormattedMessage} from 'react-intl';
import Comment from './Comment';

class CommentList extends React.Component {
  render() {
    const {section, comments, canComment, displayVisualization, isLoading} = this.props;
    const pluginContent = (
      displayVisualization ?
        this.renderPluginContent(section)
        : null
    );
    if (comments.length === 0) {
      if (!canComment || isLoading) {
        return null;  // No need to show a header for nothing at all.
      }
      return (<div className="commentlist">
        <p><FormattedMessage id="noCommentsAvailable"/></p>
      </div>);
    }
    return (<div className="commentlist">
      {pluginContent}
      {comments.map((comment) =>
        <Comment
          data={comment}
          onEditComment={this.props.onEditComment}
          onDeleteComment={this.props.onDeleteComment}
          key={comment.id + Math.random()}
          onPostVote={this.props.onPostVote}
          canVote={this.props.canVote}
        />
      )}
    </div>);
  }
}

CommentList.propTypes = {
  displayVisualization: React.PropTypes.bool,
  section: React.PropTypes.object,
  comments: React.PropTypes.array,
  canComment: React.PropTypes.bool,
  canVote: React.PropTypes.bool,
  onEditComment: React.PropTypes.func,
  onDeleteComment: React.PropTypes.func,
  onPostVote: React.PropTypes.func,
  isLoading: React.PropTypes.bool
};

export default injectIntl(CommentList);
