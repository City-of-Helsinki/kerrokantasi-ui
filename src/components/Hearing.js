import React from 'react';
import {connect} from 'react-redux';
import {push} from 'redux-router';
import Button from 'react-bootstrap/lib/Button';
import Col from 'react-bootstrap/lib/Col';
import Row from 'react-bootstrap/lib/Row';
import DeleteModal from './DeleteModal';
import {injectIntl, intlShape, FormattedMessage} from 'react-intl';
import ContactCard from './ContactCard';
import Waypoint from 'react-waypoint';

import {
  followHearing,
  postSectionComment, editSectionComment, postVote, deleteSectionComment
} from '../actions';
import SortableCommentList from './SortableCommentList';
import HearingImageList from './HearingImageList';
import LabelList from './LabelList';
import WrappedSection from './Section';
import SectionList from './SectionList';
import Sidebar from '../views/Hearing/Sidebar';
import _, {find} from 'lodash';
import Icon from '../utils/Icon';
import {
  acceptsComments,
  getClosureSection,
  getHearingURL,
  getMainSection,
  hasFullscreenMapPlugin
} from '../utils/hearing';
import {
  isSpecialSectionType,
  userCanComment,
  userCanVote
} from '../utils/section';
import getAttr from '../utils/getAttr';

export class Hearing extends React.Component {

  constructor(props) {
    super(props);

    this.state = { showDeleteModal: false, commentToDelete: { } };
  }

  openFullscreen(hearing) {
    this.props.dispatch(push(getHearingURL(hearing, {fullscreen: true})));
  }

  onPostHearingComment(text, authorName, pluginData, geojson, label, images) { // eslint-disable-line
    const {dispatch} = this.props;
    const hearingSlug = this.props.hearingSlug;
    const {authCode} = this.props.location.query;
    const mainSection = getMainSection(this.props.hearing);
    const commentData = {text, authorName, pluginData: null, authCode, geojson: null, label: null, images};
    dispatch(postSectionComment(hearingSlug, mainSection.id, commentData));
  }

  onPostSectionComment(sectionId, sectionCommentData) {
    const {dispatch} = this.props;
    const hearingSlug = this.props.hearingSlug;
    const {authCode} = this.props.location.query;
    const commentData = Object.assign({authCode}, sectionCommentData);
    dispatch(postSectionComment(hearingSlug, sectionId, commentData));
  }

  onEditSectionComment(sectionId, commentId, commentData) {
    const {dispatch} = this.props;
    const hearingSlug = this.props.hearingSlug;
    const {authCode} = this.props.location.query;
    Object.assign({authCode}, commentData);
    dispatch(editSectionComment(hearingSlug, sectionId, commentId, commentData));
  }

  onDeleteComment() {
    const {dispatch} = this.props;
    const {sectionId, commentId} = this.state.commentToDelete;
    const hearingSlug = this.props.hearingSlug;
    dispatch(deleteSectionComment(hearingSlug, sectionId, commentId));
    this.forceUpdate();
  }

  handleDeleteClick(sectionId, commentId) {
    this.setState({ commentToDelete: {sectionId, commentId}});
    this.openDeleteModal();
  }

  onVoteComment(commentId, sectionId) {
    const {dispatch} = this.props;
    const hearingSlug = this.props.hearingSlug;
    dispatch(postVote(commentId, hearingSlug, sectionId));
  }

  onFollowHearing() {
    const {dispatch} = this.props;
    const hearingSlug = this.props.hearingSlug;
    dispatch(followHearing(hearingSlug));
  }

  getFollowButton() {
    if (this.props.user === null) {
      return null;
    }
    return (
      <span className="pull-right">
        <Button bsStyle="primary" onClick={this.onFollowHearing.bind(this)}>
          <Icon name="bell-o"/> <FormattedMessage id="follow"/>
        </Button>
      </span>
    );
  }

  getClosureInfo(hearing) {
    const {formatMessage} = this.props.intl;
    const closureInfo = getClosureSection(hearing);
    if (closureInfo) {
      return closureInfo;
    }
    // Render default closure info if no custom section is specified
    return ({ type: "closure-info",
      title: "",
      abstract: "",
      images: [],
      content: formatMessage({id: 'defaultClosureInfo'}) }
    );
  }

  getLinkToFullscreen(hearing) {
    if (!hasFullscreenMapPlugin(hearing)) {
      return null;
    }
    return (
      <Button bsStyle="primary" bsSize="large" block onClick={() => this.openFullscreen(hearing)}>
        <Icon name="arrows-alt" fixedWidth/>
        <FormattedMessage id="openFullscreenMap"/>
      </Button>
    );
  }

  isSectionVotable(section, user) {
    const hearing = this.props.hearing;
    return acceptsComments(hearing) && userCanVote(user, section);
  }

  isMainSectionVotable(user) {
    const hearing = this.props.hearing;
    const section = getMainSection(hearing);
    return this.isSectionVotable(section, user);
  }

  isSectionCommentable(section, user) {
    const hearing = this.props.hearing;
    return (
      acceptsComments(hearing)
      && userCanComment(user, section)
      && !section.plugin_identifier // comment box not available for sections with plugins
    );
  }

  isMainSectionCommentable(user) {
    const hearing = this.props.hearing;
    const section = getMainSection(hearing);
    return this.isSectionCommentable(section, user);
  }

  getCommentList() {
    const {hearing, sectionComments, location, hearingSlug} = this.props;
    const mainSection = getMainSection(hearing);
    const showPluginInline = !mainSection.plugin_fullscreen;
    const user = this.props.user;
    let userIsAdmin = false;
    if (hearing && user && _.has(user, 'adminOrganizations')) {
      userIsAdmin = _.includes(user.adminOrganizations, hearing.organization);
    }
    if (!showPluginInline) {
      return null;
    }
    return (
      <div>
        <div id="hearing-comments">
          <SortableCommentList
            canVote={this.isMainSectionVotable(user)}
            displayVisualization={userIsAdmin || hearing.closed}
           section={mainSection}
           location={location}
           mainSection={mainSection}
           hearingSlug={hearingSlug}
           comments={sectionComments[mainSection.id] ?
                     sectionComments[mainSection.id].results : []}
           canComment={this.isMainSectionCommentable(hearing, user)}
           onPostComment={this.onPostHearingComment.bind(this)}
           onEditComment={this.onEditSectionComment.bind(this)}
           onDeleteComment={this.handleDeleteClick.bind(this)}
           onPostVote={this.onVoteComment.bind(this)}
           canSetNickname={user === null}
          />
        </div>
        <hr/>
      </div>
    );
  }

  openDeleteModal() {
    this.setState({ showDeleteModal: true });
  }

  closeDeleteModal() {
    this.setState({ showDeleteModal: false, commentToDelete: {} });
  }

  render() {
    const {hearing, hearingSlug, user, language, dispatch, changeCurrentlyViewed, currentlyViewed} = this.props;
    const hearingAllowsComments = acceptsComments(hearing);
    const mainSection = getMainSection(hearing);
    const showPluginInline = Boolean(!mainSection.plugin_fullscreen && mainSection.plugin_identifier);
    const closureInfoSection = this.getClosureInfo(hearing);
    const regularSections = hearing.sections.filter((section) => !isSpecialSectionType(section.type));
    const sectionGroups = groupSections(regularSections);
    return (
      <div id="hearing-wrapper">
        <LabelList className="main-labels" labels={hearing.labels}/>
        <Waypoint onEnter={() => changeCurrentlyViewed('#hearing')}/>
        <h1 className="page-title">
          {this.getFollowButton()}
          {!hearing.published ? <Icon name="eye-slash"/> : null}
          {getAttr(hearing.title, language)}
        </h1>

        <Row>
          <Sidebar
            currentlyViewed={currentlyViewed}
            hearing={hearing}
            mainSection={mainSection}
            sectionGroups={sectionGroups}
            activeLanguage={language}
            dispatch={dispatch}
            hearingSlug={hearingSlug}
          />
          <Col md={8} lg={9}>
            <div id="hearing">
              <Waypoint onEnter={() => changeCurrentlyViewed('#hearing')} topOffset={'-30%'}/>
              <div>
                <HearingImageList images={mainSection.images}/>
                <div
                  className="hearing-abstract lead"
                  dangerouslySetInnerHTML={{__html: getAttr(hearing.abstract, language)}}
                />
              </div>
              {hearing.closed ? <WrappedSection section={closureInfoSection} canComment={false}/> : null}
              {mainSection ? <WrappedSection
                showPlugin={showPluginInline}
                section={mainSection}
                canComment={this.isMainSectionCommentable(hearing, user)}
                onPostComment={this.onPostSectionComment.bind(this)}
                onPostVote={this.onVoteComment.bind(this)}
                canVote={this.isMainSectionVotable(user)}
                comments={this.props.sectionComments[mainSection.id]}
                user={user}
              /> : null}
            </div>

            {this.getLinkToFullscreen(hearing)}
            {sectionGroups.map((sectionGroup) => (
              <div id={"hearing-sectiongroup-" + sectionGroup.type} key={sectionGroup.type}>
                <Waypoint onEnter={() => changeCurrentlyViewed('#hearing-sectiongroup' + sectionGroup.name_singular)}/>
                <SectionList
                  basePath={location.pathname}
                  sections={sectionGroup.sections}
                  nComments={sectionGroup.n_comments}
                  canComment={hearingAllowsComments}
                  onPostComment={this.onPostSectionComment.bind(this)}
                  canVote={hearingAllowsComments}
                  onPostVote={this.onVoteComment.bind(this)}
                  sectionComments={this.props.sectionComments}
                  user={user}
                />
              </div>
            ))}
            <Waypoint onEnter={() => changeCurrentlyViewed('#hearing-comments')} topOffset={'-300px'}/>
            {this.getCommentList()}

            {hearing.contact_persons && hearing.contact_persons.length ?
              <h2><FormattedMessage id="contactPersons"/></h2> :
              null}
            {hearing.contact_persons && hearing.contact_persons.map((person, index) =>
              <ContactCard key={index} {...person}/> // eslint-disable-line react/no-array-index-key
            )}
          </Col>
        </Row>
        <DeleteModal
          isOpen={this.state.showDeleteModal}
          close={this.closeDeleteModal.bind(this)}
          onDeleteComment={this.onDeleteComment.bind(this)}
        />
      </div>
    );
  }
}

Hearing.propTypes = {
  intl: intlShape.isRequired,
  dispatch: React.PropTypes.func,
  hearing: React.PropTypes.object,
  hearingSlug: React.PropTypes.string,
  language: React.PropTypes.string,
  location: React.PropTypes.object,
  user: React.PropTypes.object,
  sectionComments: React.PropTypes.object,
  changeCurrentlyViewed: React.PropTypes.func,
  currentlyViewed: React.PropTypes.string
};

export function wrapHearingComponent(component, pure = true) {
  const wrappedComponent = connect((state) => ({language: state.language}), null, null, {pure})(injectIntl(component));
  // We need to re-hoist the data statics to the wrapped component due to react-intl:
  wrappedComponent.canRenderFully = component.canRenderFully;
  wrappedComponent.fetchData = component.fetchData;
  return wrappedComponent;
}

export default wrapHearingComponent(Hearing);

function groupSections(sections) {
  const sectionGroups = [];
  sections.forEach((section) => {
    const sectionGroup = find(sectionGroups, group => section.type === group.type);
    if (sectionGroup) {
      sectionGroup.sections.push(section);
      sectionGroup.n_comments += section.n_comments;
    } else {
      sectionGroups.push({
        name_singular: section.type_name_singular,
        name_plural: section.type_name_plural,
        type: section.type,
        sections: [section],
        n_comments: section.n_comments,
      });
    }
  });
  return sectionGroups;
}
