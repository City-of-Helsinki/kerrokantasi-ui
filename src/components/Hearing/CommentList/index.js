import React from 'react';
import PropTypes from 'prop-types';
import {injectIntl, FormattedMessage} from 'react-intl';
import Comment from '../Comment';

export class CommentList extends React.Component {
  render() {
    const {
      comments,
      canComment,
      isLoading,
      section,
      canReply,
      canVote,
      canFlag,
      user,
      defaultNickname,
      nicknamePlaceholder,
      language,
      hearingId,
    } = this.props;
    if (comments.length === 0) {
      if (!canComment || isLoading) {
        return null;  // No need to show a header for nothing at all.
      }
      return (
        <div className="commentlist">
          <p><FormattedMessage id="noCommentsAvailable"/></p>
        </div>
      );
    }
    return (
      <ul className="commentlist">
        {comments.map((comment) =>
          <Comment
            canReply={canReply}
            canVote={canVote}
            canFlag={canFlag}
            data={comment}
            defaultNickname={defaultNickname}
            hearingId={hearingId}
            key={comment.id + Math.random()}
            jumpTo={this.props.jumpTo}
            language={language}
            nicknamePlaceholder={nicknamePlaceholder}
            onDeleteComment={this.props.onDeleteComment}
            onEditComment={this.props.onEditComment}
            onGetSubComments={this.props.onGetSubComments}
            onPostReply={this.props.onPostReply}
            onPostVote={this.props.onPostVote}
            onPostFlag={this.props.onPostFlag}
            questions={section.questions}
            section={section}
            user={user}
          />
        )}
      </ul>
    );
  }
}

CommentList.propTypes = {
  canComment: PropTypes.bool,
  canReply: PropTypes.bool,
  canVote: PropTypes.bool,
  canFlag: PropTypes.bool,
  comments: PropTypes.array,
  defaultNickname: PropTypes.string,
  hearingId: PropTypes.string,
  isLoading: PropTypes.bool,
  jumpTo: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.number,
  ]),
  language: PropTypes.string,
  nicknamePlaceholder: PropTypes.string,
  onDeleteComment: PropTypes.func,
  onEditComment: PropTypes.func,
  onGetSubComments: PropTypes.func,
  onPostReply: PropTypes.func,
  onPostVote: PropTypes.func,
  onPostFlag: PropTypes.func,
  section: PropTypes.object,
  user: PropTypes.object,
};

CommentList.defaultProps = {
  jumpTo: -1,
};

export default injectIntl(CommentList);
