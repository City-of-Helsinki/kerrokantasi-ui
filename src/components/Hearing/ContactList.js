import React from 'react';
import PropTypes from 'prop-types';
import {Row, Col} from 'react-bootstrap';
import ContactCard from '../ContactCard';
import {FormattedMessage} from 'react-intl';

export const ContactListComponent = ({contacts, language}) => {
  return (
    <div className="hearing-contacts">
      <h3>
        <FormattedMessage id="contactPersons" />
      </h3>
      <Row>
        {contacts &&
          contacts.map((person, index) => (
            <Col
              xs={6}
              key={index + Math.random()} // eslint-disable-line react/no-array-index-key
              md={4}
            >
              <ContactCard activeLanguage={language} {...person} />
            </Col>
          ))}
      </Row>
    </div>
  );
};

ContactListComponent.propTypes = {
  contacts: PropTypes.array,
  language: PropTypes.string
};

export default ContactListComponent;
