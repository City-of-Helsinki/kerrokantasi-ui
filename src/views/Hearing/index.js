import FullscreenHearing from '../../components/FullscreenHearing';
import DefaultHearingComponent from '../../components/Hearing';
import Helmet from 'react-helmet';
import LoadSpinner from '../../components/LoadSpinner';
import React from 'react';
import {connect} from 'react-redux';
import {fetchHearing, changeCurrentlyViewed} from '../../actions';
import {getMainSection, getHearingURL, getOpenGraphMetaData} from '../../utils/hearing';
import {injectIntl} from 'react-intl';
import {push} from 'redux-router';
import getAttr from '../../utils/getAttr';


export class HearingView extends React.Component {

  constructor(props) {
    super(props);

    this.state = {currentlyViewed: 'hearing'};
    this.changeCurrentlyViewed = this.changeCurrentlyViewed.bind(this);
  }
  /**
   * Return a promise that will, as it fulfills, have added requisite
   * data for the HearingView view into the dispatch's associated store.
   *
   * @param dispatch Redux Dispatch function
   * @param getState Redux state getter
   * @param location Router location
   * @param params Router params
   * @return {Promise} Data fetching promise
   */
  static fetchData(dispatch, getState, location, params) {
    return dispatch(fetchHearing(params.hearingSlug, location.query.preview));
  }

  /**
   * Return truthy if the view can be rendered fully with the data currently
   * acquirable by `getState()`.
   *
   * @param getState State getter
   * @param location Router location
   * @param params Router params
   * @return {boolean} Renderable?
   */
  static canRenderFully(getState, location, params) {
    const {state, data} = (getState().hearing[params.hearingSlug] || {state: 'initial'});
    return (state === 'done' && data);
  }

  componentDidMount() {
    const {dispatch, params, location} = this.props;
    HearingView.fetchData(dispatch, null, location, params);
  }

  componentWillMount() {
    window.scrollTo(0, 0);
  }

  changeCurrentlyViewed(viewedItem) {
    changeCurrentlyViewed(this.props.dispatch, viewedItem);
  }

  renderSpinner() {  // eslint-disable-line class-methods-use-this
    return (
      <div className="container">
        <LoadSpinner />
      </div>
    );
  }

  checkNeedForFullscreen(hearing) {
    const fullscreenParam = this.props.location.query.fullscreen;
    const requiresFullscreen = getMainSection(hearing).plugin_fullscreen;
    if (requiresFullscreen && fullscreenParam === undefined) {
      // Looks like we need fullscreen mode, but we aren't currently using it.
      // Let's redirect to proper hearing URL
      this.props.dispatch(push(getHearingURL(hearing)));
    }
    return fullscreenParam === "true";
  }

  render() {
    const {hearingSlug} = this.props.params;
    const {state, data: hearing} = (this.props.hearing[hearingSlug] || {state: 'initial'});
    const {user, language, dispatch, currentlyViewed} = this.props;

    if (state !== 'done') {
      return this.renderSpinner();
    }

    const fullscreen = this.checkNeedForFullscreen(hearing);
    const HearingComponent = fullscreen ? FullscreenHearing : DefaultHearingComponent;

    return (
      <div key="hearing" className={fullscreen ? "fullscreen-hearing" : "container"}>
        <Helmet title={getAttr(hearing.title, language)} meta={getOpenGraphMetaData(hearing, language)} />
        <HearingComponent
          hearingSlug={hearingSlug}
          hearing={hearing}
          user={user}
          sectionComments={this.props.sectionComments}
          location={this.props.location}
          dispatch={dispatch}
          changeCurrentlyViewed={this.changeCurrentlyViewed}
          currentlyViewed={currentlyViewed}
        />
      </div>
    );
  }
}

HearingView.propTypes = {
  dispatch: React.PropTypes.func,
  hearing: React.PropTypes.object,
  params: React.PropTypes.object,
  language: React.PropTypes.string,
  location: React.PropTypes.object,
  user: React.PropTypes.object,
  sectionComments: React.PropTypes.object,
  currentlyViewed: React.PropTypes.string
};

export function wrapHearingView(view) {
  const wrappedView = connect((state) => ({
    user: state.user,
    hearing: state.hearing,
    language: state.language,
    sectionComments: state.sectionComments,
    currentlyViewed: state.hearing.currentlyViewed
  }))(injectIntl(view));

  // We need to re-hoist the data statics to the wrapped component due to react-intl:
  wrappedView.canRenderFully = view.canRenderFully;
  wrappedView.fetchData = view.fetchData;
  return wrappedView;
}

export default wrapHearingView(HearingView);
