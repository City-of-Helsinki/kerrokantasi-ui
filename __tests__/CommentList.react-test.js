import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {CommentList} from '../src/components/Hearing/CommentList';
import {MemoryRouter} from 'react-router-dom';
import {mockStore, getIntlAsProp} from '../test-utils';

// Renders the Hearings component using enzymes shallow rendering.
// You can pass props you want to override as a parameter.
const setup = propOverrides => {
  const {sectionComments, dispatch, hearing} = mockStore;
  const props = Object.assign({
    comments: sectionComments.mock.results,
    dispatch,
    canComment: true,
    isLoading: false,
    section: hearing.mockHearing.data.sections[0]
  }, propOverrides);

  const wrapper = shallow(<MemoryRouter><CommentList intl={getIntlAsProp()} {...props} /></MemoryRouter>);

  return {
    props,
    wrapper
  };
};

Math.random = jest.fn(() => 0.19859782441);

test('CommentList component should render as expected', () => {
  const {wrapper} = setup();
  const tree = toJson(wrapper.dive().dive());
  expect(tree).toMatchSnapshot();
});

test('CommentList should render a message if no comments were found', () => {
  const {wrapper} = setup({comments: []});
  const tree = toJson(wrapper.dive().dive());
  expect(tree).toMatchSnapshot();
});
