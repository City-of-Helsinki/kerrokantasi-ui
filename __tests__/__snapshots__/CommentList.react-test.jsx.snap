// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList component should render as expected 1`] = `
<ul
  className="commentlist"
>
  <InjectIntl(Comment)
    data={
      {
        "answers": [
          {
            "answers": [
              115,
            ],
            "question": 6,
            "type": "single-choice",
          },
        ],
        "author_name": "Mikko Uro",
        "can_edit": false,
        "content": "Testi",
        "created_at": "2017-10-23T11:23:47.475069Z",
        "geojson": null,
        "hearing": "rXT2L1HxEOZTERjluyxyQZ412aYM8oZE",
        "id": 2615,
        "images": [],
        "is_registered": true,
        "label": null,
        "language_code": "",
        "n_votes": 0,
        "plugin_data": "",
        "section": "7CxRxrlvrW9uCCjEaB9rW1dENHQMudeK",
      }
    }
    jumpTo={-1}
    key="2615.19859782441"
    questions={
      [
        {
          "id": 6,
          "is_independent_poll": false,
          "n_answers": 1,
          "options": [
            {
              "id": 115,
              "n_answers": 1,
              "text": {
                "en": "NO",
                "fi": "EI !",
              },
            },
            {
              "id": 116,
              "n_answers": 0,
              "text": {
                "en": "Yes",
                "fi": "Joo",
              },
            },
          ],
          "text": {
            "en": " do you like music ?",
            "fi": "Pidätkö sinä musiikista ?",
          },
          "type": "single-choice",
        },
      ]
    }
    section={
      {
        "abstract": {
          "fi": "Tämä on suomenkielinen testiabstrakti. Sisältää huikean kuvauksen kyseisestä kuulemisesta.",
        },
        "commenting": "open",
        "content": {
          "fi": "<p>Samanaikaisesti Isosaaren avaamisen kanssa Helsingin kaupunki aloittaa saaren asemakaavoituksen. Saarta suunnitellaan monipuoliseksi virkistyksen ja matkailun keitaaksi. &nbsp;&nbsp;&nbsp;</p>
<p>Kerro ajatuksesi Isosaaren kehittämisestä: &nbsp;</p>
<ul>
  <li>Minkälaisia matkailupalveluja ja toimintaa Isosaaressa voisi olla?</li>
  <li>Mikä Isosaaressa on ainutlaatuista?&nbsp;</li>
  <li>Millainen Isosaari olisi tulevaisuudessa? Ideoita ja visioita käytetään saaren suunnittelun tukena ja innoituksena.&nbsp;</li>
</ul>
<p>&nbsp;&nbsp;Ideoita ja visioita käytetään saaren suunnittelun tukena ja innoituksena.</p>
<p>Lisätietoja:</p>
<p><a href="http://www.visitisosaari.fi">Tietoa Isosaaressa vieraileville</a></p>",
        },
        "created_at": "2017-10-23T11:07:26.668934Z",
        "id": "7CxRxrlvrW9uCCjEaB9rW1dENHQMudeK",
        "images": [
          {
            "caption": {
              "fi": "A Wonderfull caption",
            },
            "height": 631,
            "id": 92,
            "title": {},
            "url": "https://api.hel.fi/kerrokantasi-test/media/images/2017/10/LzqVMrMI.jpeg",
            "width": 893,
          },
        ],
        "n_comments": 1,
        "plugin_data": "",
        "plugin_fullscreen": false,
        "plugin_identifier": "",
        "plugin_iframe_url": "",
        "published": false,
        "questions": [
          {
            "id": 6,
            "is_independent_poll": false,
            "n_answers": 1,
            "options": [
              {
                "id": 115,
                "n_answers": 1,
                "text": {
                  "en": "NO",
                  "fi": "EI !",
                },
              },
              {
                "id": 116,
                "n_answers": 0,
                "text": {
                  "en": "Yes",
                  "fi": "Joo",
                },
              },
            ],
            "text": {
              "en": " do you like music ?",
              "fi": "Pidätkö sinä musiikista ?",
            },
            "type": "single-choice",
          },
        ],
        "title": {},
        "type": "main",
        "type_name_plural": "pääosiot",
        "type_name_singular": "pääosio",
        "voting": "registered",
      }
    }
    showReplies={false}
  />
</ul>
`;

exports[`CommentList should render a message if no comments were found 1`] = `
<div
  className="commentlist"
>
  <p>
    <FormattedMessage
      id="noCommentsAvailable"
      values={{}}
    />
  </p>
</div>
`;
